packages/mobile/src/firebase/notifications.ts

```diff
@@ -73,20 +75,20 @@ const handlePaymentReceived = (
   }
 }
 
-export const handleNotification = (
+export function* handleNotification(
   notification: Notification,
   notificationState: NotificationReceiveState
-) => async (dispatch: DispatchType, getState: GetStateType) => {
+) {
   if (notificationState === NotificationReceiveState.APP_ALREADY_OPEN) {
-    dispatch(showMessage(notification.title))
+    yield put(showMessage(notification.title))
   }
   switch (notification.data.type) {
     case NotificationTypes.PAYMENT_REQUESTED:
-      dispatch(handlePaymentRequested(notification.data, notificationState))
+      yield handlePaymentRequested(notification.data, notificationState)
```

Comment:
Should this be yield put?

Reply:
`handlePaymentRequested ` is a generator, should be fine, right?

Reply:
If it's an action to dispatch, you should `yield put`. If it's a function to call, you should `yield call`.
`yield` alone may work if the function is a generator, but it improves testability and error handling to use yield call